compiler/noirc_frontend/src/parser/parser/function.rs

```diff
@@ -76,16 +79,32 @@ fn function_modifiers() -> impl NoirParser<(bool, ItemVisibility, bool)> {
         })
 }
 
+pub(super) fn numeric_generic() -> impl NoirParser<UnresolvedGeneric> {
+    keyword(Keyword::Let)
+        .ignore_then(ident())
+        .then_ignore(just(Token::Colon))
+        .then(parse_type())
+        .map(|(ident, typ)| UnresolvedGeneric::Numeric { ident, typ })
+}
+
+pub(super) fn generic_variable() -> impl NoirParser<UnresolvedGeneric> {
```

Comment:
Nitpick: both normal & numeric generics are generic variables so we should rename this to `non_numeric_generic` or similar

Reply:
I went with `generic_type`. What do you think?